Update to the doc removing the IE8 support

GianlucaGuarini opened the pull request:
@tipiirai @cognitom @rsbondi @aMarCruz 
I am ready to release riot 2.2.2 We should just update a bit the documentation.
Please let me know what you think about my updates. I was thinking also about a script to replace the riot size in automatically on any new release

rsbondi commented on doc/compiler.md:
```diff
@@ -55,9 +55,7 @@ riot.compile(function() {
-The compiler weights only 3.2KB (1.7K gzipped) so you can safely perform client side compilation on production without download or performance or issues.
-
-Just like Riot itself the compiler works on IE8 as well.
+The compiler weights so you can safely perform client side compilation on production without download or performance or issues.
```
'The compiler is lightweight' instead of 'The compiler weights' (that should have been weighs)

r4j4h commented:
https://www.youtube.com/watch?v=WChTqYlDjtI R.I.P. IE8

r4j4h commented on doc/faq.md:
```diff
@@ -11,10 +11,6 @@ description: none
 
-### Why IE8?
-Because it's widely used. According to [Net Market Share](http://www.netmarketshare.com/) the global desktop market share is 19% and according to [StatCounter](http://statcounter.com/demo/browser/) the market share is 2.6%.
-So Please consider [IE8 support](https://muut.com/riotjs/download.html#ie8-support) on your project.
```
If IE8 support is being fully dropped we should clean up the [linked segment on download page](https://muut.com/riotjs/download.html#ie8-support) - or re-add this line with a note that IE8 is not actually supported and will require more work (shims, coding style modifications, etc) or whatever is needed

aMarCruz commented:
great work!! :+1:

GianlucaGuarini commented:
@r4j4h :smile_cat: yeah we should remove also the ie8 stuff in the download section
@rsbondi thanks for your input

GianlucaGuarini commented:
@tipiirai I have updated the doc using the size of the latest riot, and removed any reference to the size and IE8. I've also added a file to make this process automatic on any new release `$ node doc/update-riot-size`
https://github.com/riot/riot/blob/update/doc/doc/update-riot-size.js

Let me know if it's ok for you and I will merge this pull request to make a new release

a-moses commented:
Search engine : The Description and Titles  Needs improvement and to be  updated to latest  tag-line 
![riotjs-google](https://cloud.githubusercontent.com/assets/11050286/8635555/754aa92a-2817-11e5-92ca-2dee956dc3a9.jpg)

anthias commented:
No matter how much we hate IE8, it seems completely inappropriate to drop support in a minor version bump. Looking at the process of rolling out 2.2, this seems like a side effect and was never really discussed https://github.com/riot/riot/issues/773. Now this "documentation change" is holding up real bugs on 2.2.1. Does anyone else see this as an issue?

GianlucaGuarini commented:
We do not drop ie8, if you have used riotjs on this browser with all the polyfills/hacks until now it and your project is still running somehow on that browser, nothing will change. This update is needed to remove all the documentation about ie8 because __riot has never passed the tests on ie8 since the 2.0.15 release__ and it's not fair to push its compatibility on this old browser as one of its strongest advantage.
__I can not guarantee that riot works on IE8 without test__
If you guys ( @tipiirai @cognitom @rsbondi @aMarCruz  ) think that riot is still a framework to develop apps on IE8 I will close this issue and roll out a new release otherwise let's merge this and move forward.

@anthias If you know an easy way to fix all the tests on IE8 I am more than happy, but I can tell you that I have already tried everything with no big success. Pull requests are welcome

anthias commented:
@GianlucaGuarini I see your point and sympathize with you on the development front. My comment was specifically around the process. It sounds like this is an issue (limited support for IE8) that has been around for some time and you're simply addressing it.

My request, as someone who is looking to implement riot with our customers, is that support is clearly maintained or not maintained. Of course, choosing to not support something should be explicitly communicated too (e.g. change log, version bump etc.).

Thank you for the hard work on this project!

GianlucaGuarini commented:
In riot we like democracy ( or at least I do ) and I appreciate any feedback to improve it, so thanks @anthias for your clever post! Let's see what's the @tipiirai's opinion. In the worst case I will publish a new release keeping this issue opened for further discussions.

avimar commented:
The website lists riot 2.2.1 and github still shows the ie8 text.. I'm confused, did you finish cutting the new release?

cognitom commented:
Hi, I've been catching up this issue. (last week I was too busy for my client works.. >< )

I can fix the issue about style injection on IE8 with a simple polyfill around this line.
https://github.com/riot/riot/blob/master/lib/browser/tag/vdom.js#L17

BTW, I'm not sure it's good or bad to keep support of ancient IEs. At least, we should separate the code and polyfill, I think.

cognitom commented:
@GianlucaGuarini @anthias I've been using Riot.js mainly for the business application, so IE support didn't matter for me. But I have a plan to extend it to the other public web services, too. We have 1.6% IE8 users on that. I'm not sure we can ignore the 1.6%..., but basically we'll prefer to focus on the mobiles (not PC).

Anyway I've made a report about ie hacks on Riot.js, kind of :) #983

GianlucaGuarini commented:
If someone will find a way to run all the riot unit test successfully on ie8 we could keep the support, otherwise we are just lying

avimar commented:
Perhaps use https://www.browserstack.com? It says "Free for Open Source"
and they have IE8. Saucelabs doesn't have IE8? I must say their website is
fancy but short on actual details.

tipiirai commented:
I just updated the website with v2.2.2. Sorry – I was offline on the countryside.

I think we should simply drop IE8 alltogether. Too much trouble for everyone.

Mouvedia commented:
I totally disagree. IE8 needs to be supported. It's still popular and decent enough.
I want to use riot in my next project.

We can discuss how it should be supported though.
Shims and PRs are probably the way to go.

tipiirai commented:
Unfortunately IE8 support is simply too expensive in terms of development time and the resulting file size. As much as we all like to support IE8 – the benefit doesn't outweigh all the trouble.

tipiirai commented:
The slope of IE8 is firmly going down to zero. We don't want to waste huge efforts for the next ~1.5 years or so.

![image](https://cloud.githubusercontent.com/assets/169330/8893019/2e6dac5c-337d-11e5-89a4-1fac74050069.png)

Mouvedia commented:
If you remove fixes for IE8, add them to a separate file.
We need to be able to build a IE8 compatible version of riot if our top priority is not the size.

tipiirai commented:
Unfortunately IE8 must go, mostly because it causes so much pain for our development workflow. We  don't want to slow down the progress and innovation because one dying browser. This is definitely not a light decision. 

File size is just a minor thing compared to above.

GianlucaGuarini commented:
this pull request is old and we moved the doc to `riot.github.io`
